components/mailchimp/sources/new-or-updated-list-segment/new-or-updated-list-segment.js

```diff
@@ -0,0 +1,110 @@
+const common = require("../common/timer-based");
+
+module.exports = {
+  ...common,
+  key: "new-or-updated-list-segment",
+  name: "New Or Updated List Segment",
+  description: "Emit new event when segment is either created or updated.",
+  version: "0.0.1",
+  type: "source",
+  dedupe: "unique",
+  props: {
+    ...common.props,
+    listId: {
+      propDefinition: [
+        common.props.mailchimp,
+        "listId",
+      ],
+      description:
+      "The unique ID of the audience list which you'd like to watch for new or updated segments.",
+    },
+    watchFor: {
+      type: "string",
+      label: "Watch for new created or updated segments?",
+      description:
+        "If set to `Created`, it will include new created segments only.  When set to `Updated`, it will only include recently updated segments.",
+      options: [
+        "Created",
+        "Updated",
+      ],
+      optional: false,
+    },
+    includeTransactional: {
+      propDefinition: [
+        common.props.mailchimp,
+        "includeTransactional",
+      ],
+    },
+  },
+  hooks: {
+    async deploy() {
+      // Emits sample events on the first run during deploy.
+      const config = {
+        count: 10,
+        offset: 0,
+      };
+     let segments;
+     if(this.watchFor === "Created" ) {
+      segments = await this.mailchimp.getAudienceSegmentsByCreatedDate(this.listId, config);
+     } else {
+      segments = await this.mailchimp.getAudienceSegmentsByUpdatedDate(this.listId, config);
+     }
```

Comment:
Could simplify it this way:
```suggestion
     const segments = this.watchFor === "Created"
        ? await this.mailchimp.getAudienceSegmentsByCreatedDate(this.listId, config)
        : segments = await this.mailchimp.getAudienceSegmentsByUpdatedDate(this.listId, config);
```